python/cuspatial/cuspatial/tests/binpreds/binpred_test_dispatch.py

```diff
@@ -0,0 +1,535 @@
+        """
+    -----
+    | x |
+    | | |
+    | x |
+    -----
+    """,
+        LineString([(0.5, 0.25), (0.5, 0.75)]),
+        point_polygon,
+    ),
+    "linestring-polygon-in-out": (
+        """
+    -----
+    |   |
+    | x |
+    | | |
+    --|--
+      |
+      x
+    """,
+        LineString([(0.5, 0.5), (0.5, -0.5)]),
+        point_polygon,
+    ),
+    "linestring-polygon-crosses": (
+        """
+      x
+    --|--
+    | | |
+    | | |
+    | | |
+    --|--
+      x
+    """,
+        LineString([(0.5, 1.25), (0.5, -0.25)]),
+        point_polygon,
+    ),
+    "polygon-polygon-disjoint": (
+        """
+    Polygon polygon tests use a triangle for the lhs and a square for the rhs.
+    The triangle is drawn as
+    x---x
+    |  /
+    | /
+    |/
+    x
+
+    The square is drawn as
+
+    -----
+    |   |
+    |   |
+    |   |
+    -----
+    """,
+        Polygon([(0.0, 2.0), (0.0, 3.0), (1.0, 3.0)]),
+        point_polygon,
+    ),
+    "polygon-polygon-touch-point": (
+        """
+    x---x
+    |  /
+    | /
+    |/
+    x----
+    |   |
+    |   |
+    |   |
+    -----
+    """,
+        Polygon([(0.0, 1.0), (0.0, 2.0), (1.0, 2.0)]),
+        point_polygon,
+    ),
+    "polygon-polygon-touch-edge": (
+        """
+     x---x
+     |  /
+     | /
+     |/
+    -x--x
+    |   |
+    |   |
+    |   |
+    -----
+    """,
+        Polygon([(0.25, 1.0), (0.25, 2.0), (1.25, 2.0)]),
+        point_polygon,
+    ),
+    "polygon-polygon-overlap-edge": (
+        """
+    x
+    |\\
+    | \\
+    |  \\
+    x---x
+    |   |
+    |   |
+    |   |
+    -----
+    """,
+        Polygon([(0.0, 1.0), (0.0, 2.0), (1.0, 2.0)]),
+        point_polygon,
+    ),
+    "polygon-polygon-point-inside": (
+        """
+      x---x
+      |  /
+      | /
+    --|/-
+    | x |
+    |   |
+    |   |
+    -----
+    """,
+        Polygon([(0.5, 0.5), (0.5, 1.5), (1.5, 1.5)]),
+        point_polygon,
+    ),
+    "polygon-polygon-point-outside": (
+        """
+     x
+    -|\\--
+    |x-x|
+    |   |
+    |   |
+    -----
+    """,
+        Polygon([(0.25, 0.75), (0.25, 1.25), (0.75, 0.75)]),
+        point_polygon,
+    ),
+    "polygon-polygon-in-out-point": (
+        """
+      x
+      |\\
+    --|-x
+    | |/|
+    | x |
+    |   |
+    x----
+    """,
+        Polygon([(0.5, 0.5), (0.5, 1.5), (1.0, 1.0)]),
+        point_polygon,
+    ),
+    "polygon-polygon-in-point-point": (
+        """
+    x----
+    |\\  |
+    | x |
+    |/  |
+    x----
+    """,
+        Polygon([(0.0, 0.0), (0.0, 1.0), (0.5, 0.5)]),
+        point_polygon,
+    ),
+    "polygon-polygon-contained": (
+        """
+    -----
+    |  x|
+    | /||
+    |x-x|
+    -----
+    """,
+        Polygon([(0.25, 0.25), (0.75, 0.75), (0.75, 0.25)]),
+        point_polygon,
+    ),
+    "polygon-polygon-same": (
+        """
+    x---x
+    |   |
+    |   |
+    |   |
+    x---x
+    """,
+        point_polygon,
+        point_polygon,
+    ),
```

Comment:
@harrism the "Documentation" is primarily here. I can move it to the binary predicates design document if you like, as well.

Reply:
OK!